### PR TITLE
fix: git discard message

### LINKED ISSUE #24946: [Bug]: Discard message's subject is missing when we only have navigation changes/ theme settings
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Current message:
Screenshot 2023-06-30 at 11 39 51 AM

Expected message:
Screenshot 2023-06-30 at 11 41 03 AM

### Steps To Reproduce

1. Go to git connected app
2. Go to app settings, change navigation
3. Open git sync modal and click on discard button
4. Observe the message is missing the subject

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by brayn003**
## Description
Discard message were conditionally not shown. It caused a bug when only navigation settings were changed. Removed the condition to include the message every-time.

#### PR fixes following issue(s)
Fixes #24946 

#### Media

Before
image

After
image

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by hetunandu — DISMISSED** [review]

File: app/client/cypress/support/Objects/CommonLocators.ts [hunk 1]
```diff
@@ -250,4 +250,6 @@ export class CommonLocators {
   _gitStatusChanges = "[data-testid='t--git-change-statuses']";
   _appNavigationSettings = "#t--navigation-settings-header";
   _appNavigationSettingsShowTitle = "#t--navigation-settings-application-title";
+  _gitDiscardBtn = ".t--discard-button";
+  _gitDiscardCallout = "[data-testid='t--discard-callout']";
 }
```
Comment (Parthvi12) [inline app/client/cypress/support/Objects/CommonLocators.ts:255]: @brayn003 _gitDiscardBtn locator exist in GitSync.ts, can we use it from there, also please move _gitDiscardCallout to GitSync.ts too.

Comment (brayn003) [reply app/client/cypress/support/Objects/CommonLocators.ts]: @Parthvi12 this is resolved

**Review by Parthvi12 — APPROVED** [review]

**Pull request merged**